recirq/otoc/loschmidt/tilted_square_lattice/analysis-walkthrough.ipynb

```diff
@@ -0,0 +1,373 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "import matplotlib\n",
+    "matplotlib.rcParams.update(**{\n",
+    "    'axes.titlesize': 14,\n",
+    "    'axes.labelsize': 14,\n",
+    "    'xtick.labelsize': 12,\n",
+    "    'ytick.labelsize': 12,\n",
+    "    'legend.fontsize': 12,\n",
+    "    'legend.title_fontsize': 12,\n",
+    "    'figure.figsize': (7, 5),\n",
+    "})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from recirq.otoc.loschmidt import TiltedSquareLatticeLoschmidtSpec\n",
+    "\n",
+    "assert TiltedSquareLatticeLoschmidtSpec, 'register deserializer'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load Raw Results\n",
+    "\n",
+    "After some basic imports, we load the `ExecutableGroupResult` in its entirety using `cirq_google.workflow` tools. This is the complete raw dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "import cirq_google as cg\n",
+    "\n",
+    "import recirq.otoc.loschmidt.tilted_square_lattice.analysis as analysis"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "raw_results = cg.ExecutableGroupResultFilesystemRecord.from_json(run_id='simulated-1').load()\n",
+    "repr(raw_results)[:100] + ' ...'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use Pandas for slicing and dicing\n",
+    "\n",
+    "We can extract the most relevant parameters from the raw results and flatten it into a tabular format. In particular, we use `pd.DataFrame` for further data aggregation and plotting.\n",
+    "\n",
+    "The measured `success_probability` ranges between 0 and 1, indicating the fraction of times the measured result after the echo actually returned to its initial state."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = analysis.loschmidt_results_to_dataframe(raw_results)\n",
+    "print(len(df), 'rows')\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Aggregation\n",
+    "\n",
+    "Instead of considering each result in isolation, we can aggregate quantities to make more meaningful plots.\n",
+    "\n",
+    "In the following, we use pandas group-by functionality to\n",
+    " 1. Average (and compute the standard deviation) over random instances, holding all else constant.\n",
+    " 2. Plot these averaged quantities vs. macrocycle_depth, holding all else constant.\n",
```

Comment:
Could you elaborate on why these aggregations are useful for the experiment? Or would the target audience have enough context to understand?

Reply:
added "Instead of considering each result in isolation, we can aggregate quantities to make more meaningful plots **and fits**."

I think the audience should have enough context by the time they get to this "analsys" notebook. That sort of explanation is probably appropriate for an introduction to the loschmidt echo benchmark which can be done in follow-on work to provide better documentation.